src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/PackageMapReferenceAnalyzer.cs

```diff
@@ -68,12 +68,24 @@ public async Task AnalyzeAsync(IProject project, IDependencyAnalysisState state,
                 foreach (var map in packageMaps.Where(m => m.ContainsAssemblyReference(reference.Name)))
                 {
                     _logger.LogInformation("Marking assembly reference {ReferenceName} for removal based on package mapping configuration {PackageMapSet}", reference.Name, map.PackageSetName);
-                    state.References.Remove(reference, new OperationDetails() { Risk = BuildBreakRisk.Medium });
+                    state.References.Remove(reference, new OperationDetails() { Details = ProcessReferenceWarnings(map.PackageSetWarning), Risk = BuildBreakRisk.Medium });
                     await AddNetCoreReferences(map, state, token).ConfigureAwait(false);
                 }
             }
         }
 
+        private IEnumerable<string> ProcessReferenceWarnings(string warning)
```

Comment:
This could be done in the packages.remove call (see https://github.com/dotnet/upgrade-assistant/issues/765 for an existing issue).

Reply:
Makes sense, as part of #765 we can unify the log statements to happen along with add or remove.